packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -467,22 +474,26 @@ export function checkProofListMembershipInputsForConfig(
 
           if (!_.isEqual(elementWidth, comparisonWidth)) {
             throw new TypeError(
-              `Membership list ${listName} in input contains element of width ${JSON.stringify(
-                elementWidth
-              )} while comparison value with identifier ${JSON.stringify(
+              `Membership list ${listName} in input contains element of width ${elementWidth} while comparison value with identifier ${JSON.stringify(
                 comparisonId
-              )} has width ${JSON.stringify(comparisonWidth)}.`
+              )} has width ${comparisonWidth}.`
             );
           }
         }
 
-        // The comparison value should lie in the membership list.
+        // The comparison value should lie in the membership list. We relax the
```

Comment:
The comment about relaxed type checking isn't really relevant to the hash-based comparison.  Instead I suggest a comment explaining why comparison is by hash.